### PR TITLE
fix(web)[TextInput]: cursor jumps to the start when secureTextEntry i…

### LINKED ISSUE #21727: [$1000] Chat - Tapping eye icon for protected pdf file moves cursor to the start of the field
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/21367
## Action Performed:
1. Login to NewDot
2. Open chat
3. Tap password protected pdf
4. Tap enter password
5. Type several characters
6. Tap Eye icon

## Expected Result:
Cursor should stay where it was

## Actual Result:
Cursor moves to the start fo the field

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.33.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/59bc8698-7c3b-421a-a795-620d982f80a7

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f00fa406bef36c56</li>
        <li>Upwork Job ID: 1674042523612561408</li>
        <li>Last Price Increase: 2023-06-28</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

### PR DESCRIPTION
**Pull request by pac-guerreiro**
This PR fixes https://github.com/Expensify/App/issues/21727

# Problem

On web, the cursor jumps to the start of the input when `secureTextEntry` is toggled.

# Solution

Preserve and restore previous selection after `secureTextEntry` is toggled.

The proposal and demo are available in this comment: https://github.com/Expensify/App/issues/21727#issuecomment-1642507490

### CODE REVIEW

File: packages/react-native-web/src/exports/TextInput/index.js [hunk 1]
```diff
@@ -323,18 +330,22 @@ const TextInput: React.AbstractComponent<
...
-      try {
-        const node = e.target;
-        const { selectionStart, selectionEnd } = node;
-        e.nativeEvent.selection = {
-          start: selectionStart,
-          end: selectionEnd
-        };
+    try {
+      const node = e.target;
+      const { selectionStart, selectionEnd } = node;
```
Comment (thiagobrez) [inline packages/react-native-web/src/exports/TextInput/index.js]: nitpick: I would do this in a single line

```suggestion
      const { selectionStart, selectionEnd } = e.target;
```

**Review by sobitneupane — COMMENTED** [review]
@pac-guerreiro In IOS/Safari, I am having following issue

https://github.com/Expensify/react-native-web/assets/25876548/2a8791ef-bf1f-46a6-aee3-d8b67e79be05

**Comment by pac-guerreiro** [comment]
Uploading Screen Recording 2023-08-01 at 18.46.43.mov…

@sobitneupane 

You sure you're running my code? I just tested my changes on iOS/safari and they work 😅

**Comment by sobitneupane** [comment]
@pac-guerreiro Looks like the screen recording didn't get uploaded in previous message

Screenshot 2023-08-02 at 13 30 18

**Comment by pac-guerreiro** [comment]
@sobitneupane 

My bad 😅 

https://github.com/Expensify/react-native-web/assets/48553277/685c7cce-e09c-40a7-8ead-e8830e13eb8a

**Comment by sobitneupane** [comment]
Thanks @pac-guerreiro.

We are good to merge this.

cc: @NikkiWines

**Review by sobitneupane — APPROVED** [review]

**Comment by NikkiWines** [comment]
Looks like there are some formatting and unit test errors

**Comment by pac-guerreiro** [comment]
@NikkiWines 

My bad 😅 I just pushed fixes for those issues!

**Comment by NikkiWines** [comment]
No worries @pac-guerreiro - looks like there's still one failure

**Comment by pac-guerreiro** [comment]
@NikkiWines second time's the charm 😅

**Review by NikkiWines — APPROVED** [review]
:shipit:

**Comment by roryabraham** [comment]
Hello everyone 👋🏼 few PSA's here:

- Generally we should not review or merge any PRs in any of our forked dependencies if there is not an equivalent PR in the upstream. Same goes for any PR that introduces a patch file
- I think we are going to stop using this fork and move to patch files instead for all the same reasons we ditched the react-native fork. This repo is easier to build and publish, but still it's an extra step we don't want to deal with and we invested in an [awesome new feature in patch-package](https://github.com/ds300/patch-package/issues/43) that allows us to have multiple patch files for a single dependency applied in sequence.

So in conclusion, I think we need to:

- create an issue in the upstream react-native-web repo with a minimal reproduction of the bug
- create an upstream PR in the react-native-web repo for this change 
- Only once that's completed and we have an upstream PR to reference should we proceed with this code in a patch file in E/App instead of in this fork
- To prevent any duplicate work, we should wait until https://github.com/Expensify/App/issues/16660 is complete before introducing the new patch including this code
